framework/web/Response.php

```diff
@@ -488,6 +485,9 @@ public function sendContentAsFile($content, $attachmentName, $options = [])
             $this->content = $content;
         }
 
+        $mimeType = isset($options['mimeType']) ? $options['mimeType'] : 'application/octet-stream';
+        $this->setDownloadHeaders($attachmentName, $mimeType, !empty($options['inline']), $end - $begin + 1);
```

Comment:
Input is `$content`, `$end < StringHelper::byteLength($content)`, so input is always long enough